qt/applications/workbench/workbench/plotting/figuremanager.py

```diff
@@ -237,15 +240,24 @@ def __init__(self, canvas, num):
         # resize the main window so it will display the canvas with the
         # requested size:
         cs = canvas.sizeHint()
+        ts = self.toolbar.sizeHint()
         sbs = self.window.statusBar().sizeHint()
         self._status_and_tool_height = tbs_height + sbs.height()
         height = cs.height() + self._status_and_tool_height
-        self.window.resize(cs.width(), height)
+        # get window width from toolbar/canvas width
+        if cs.width() > ts.width():
```

Comment:
This works, but I wonder if one should adjust also the height to keep the aspect ratio the same.
```suggestion
 width = max(cs.width(), ts.width())
```

Reply:
I changed the if else statement with max(). For the aspect ratio, i do not understand which one. The window height comes from the canvas + the toolbar, and the width from the max between the two widgets. Which ratio should be kept constant?

Reply:
I think the canvas aspect ratio should be kept constant for the time being.
Still it is dangerous to take the max of the toolbar as it can potentially grow further.
So the best would be to make them a user setting, so one can customize also based on the screen size and resolution. I've opened another issue for this: https://github.com/mantidproject/mantid/issues/31724

Reply:
This is what I get now. There is a fair bit of empty space in the toolbar, even though we take its size. Any idea?
![Screenshot 2021-06-07 at 10 05 21](https://user-images.githubusercontent.com/16865759/120981395-eca1e280-c777-11eb-8d2b-96c6ac22efad.png)